More stuff

Hellowlol opened the pull request:
So this PR got alot bigger then what i indended.

Anyways this adds support for more controls and make this easier to work with. Everything is a class and all the props are hardcoded as i didnt like the default name that tesla uses or it returns some kind of info that isnt very user friendly. (center_display_state value 8 == dog mode) etc.

This is a wip, need some cleanup, add more logging and tests.

Hellowlol committed:
Zomg, this got much bigger then intented

Hellowlol committed:
stuff

Dreamsorcerer commented:
That is big. :P
I'll try and take an initial look through in a few days. Note that it will be easier if you don't make a lot of formatting changes (maybe you were running Black or something automatically?).

Hellowlol commented:
Yeah, its black. It runs each time I hit save 🤦‍♂️. I’ll disable it now. 

I’ll add some tests and examples on why I think this Is a better solution. The goal is to have a user friendly python classes while still able to use the raw json response if needed.

I’m not sure how to add the test as I have never used azure pipelines before, but I’ll figure that out.

Hopefully I’ll  have this ready over the weekend, I just wanted to get some kinda feedback. 


Thanks!

Dreamsorcerer commented:
I think @mlowijs was experimenting with the Azure Pipelines, but I don't think he finished setting it up.

So, just ignore that. As long as I can run the tests locally, that should be enough. Though feel free to setup CI integration with Travis or something similar.

Hellowlol committed:
Add tests for the class attributes.

Dreamsorcerer commented on tesla_api/misc.py:
```diff
@@ -0,0 +1,30 @@
+def cast(value):
+    """Helper"""
+    if isinstance(value, str):
+        return False if value == "0" else True
```
```suggestion
        return value != "0"
```

Dreamsorcerer commented on tesla_api/misc.py:
```diff
@@ -0,0 +1,30 @@
+    if isinstance(value, str):
+        return False if value == "0" else True
+    if isinstance(value, int):
+        return False if value is 0 else True
```
```suggestion
        return value != 0
```
Don't use `is`. It`ll probably work, but it's not guaranteed (it's implementation dependant).

Dreamsorcerer commented on tesla_api/speedlimit.py:
```diff
@@ -0,0 +1,60 @@
+        return self._vehicle._format_distance_unit(self.max_limit_mph)
+
+    @property
+    def pin_clode_set(self):
```
```suggestion
    def pin_code_set(self):
```

Dreamsorcerer commented:
I'll try and take a broader look over the weekend and see if I can get an understanding of it.

Note that this may intersect with my changes in https://github.com/mlowijs/tesla_api/pull/33/files, which I started sometime ago, but haven't got around to finishing off. That PR adds very strict typing throughout the full code base, enforced with Mypy.

Hellowlol committed:
Change some method names

== != is
add locked property

Hellowlol commented:
Thats nice, i plan to use typing, i started on the low hanging fruit. But i got bored. I'll run monkeytype when after all the tests are done.

Hellowlol committed:
patch _command

Hellowlol committed:
Fix typos

Fix some regarding feedback.
Add some more methods to the tests

Hellowlol committed:
Add test

Hellowlol committed:
Add dev install option

Change fix failing test.

Hellowlol committed:
First pass on the ci

Hellowlol committed:
Some docstring and test.

Hellowlol committed:
Add tests for the rest of the climate commands

Hellowlol committed:
Add more test

Many need to be fixed with patches.

Hellowlol committed:
more tests.

Hellowlol committed:
Update ci.yaml

Hellowlol committed:
Add codecov and remove pipelines

Hellowlol committed:
Create pypi.yaml

Hellowlol committed:
Add missing pytest-asyncio

tp setup.py

Hellowlol committed:
Update ci.yaml

Hellowlol committed:
Add nissing pytest cov

Hellowlol committed:
Update ci.yaml

Hellowlol committed:
Windows tests

Hellowlol committed:
Fix typo and add CLIMATE_STATE to __all__

Hellowlol committed:
add test for battery

Hellowlol committed:
add test for config

Hellowlol committed:
Add tests for media

Hellowlol committed:
Add sentry tests.

Hellowlol committed:
misc and status tests.

Hellowlol committed:
Pushing coverage

Hellowlol committed:
Fix ImportError

Hellowlol committed:
Create test_controls.py

Hellowlol committed:
Delete test_misc.py

Copy pasta error.

Hellowlol committed:
Add software file.

Hellowlol committed:
Add software to controls.

Hellowlol committed:
Add software tests

Hellowlol committed:
Fix typo so test pass

Hellowlol committed:
Remove debug log from the test writer

and run black

Hellowlol commented:
Ok, this **_beast_** should the ready for renew, it isn't done but I suspect it will take some time to do the review.

The goal of the PR is to have classes that hides the logic for what value to find and how to format it.
Fx is the speedlimit requires the limit in mile per hour is will check what unit type the car uses and convert if needed. Same goes for other attributes where the api returns the data in another format then what the car uses.

Raw json response supported via ``` vehicle.get_charge_state``` etc

Removed:
 - azure_piplines (seemed to be broken anyway)

Added/Improved:
- Github action is now used for CI
- 75 test written (and counting)
- 86 % coverage, i want to cover everything not energy related. https://codecov.io/gh/Hellowlol/tesla_api/branch/more-controls
- package upload

TODO:
Readme will updated when the PR has been approved.
Update the doc string to better explain what this property mean/does

And down the road:
- lint with black
- Add typing and lint with mypy
- Add docs to readthedocs

Hellowlol committed:
Fix passenger typo

Dreamsorcerer commented:
OK, I'll take a look this week. Let's see if we can break it up into some smaller pieces as we go.

Hellowlol committed:
annotations

Hellowlol committed:
Add tool update FULL_DATA .

Only handles added and removed keys, as changes would require a full rewrite.

Hellowlol committed:
Monkeytype can eat a banana!

Hellowlol committed:
Typing..

Hellowlol committed:
Fix battery bug and add typing to controls

Hellowlol committed:
Add typing to media

Hellowlol committed:
Add typing to sentry.py

Hellowlol committed:
Add typing to software.py

Hellowlol committed:
Add typing to speedlimit.py

Dreamsorcerer commented:
Don't bother with conflicts right now. I'm going to merge the Mypy stuff first, which will just create more conflicts.

Hellowlol commented:
Ok. Ill still add the typing as I need this to create the docs. Im to lazy to write them myself.

Pretty awsome how much its possible to semi automate. Use a tool that creates the tests that creates typing hints that is used to create the docs 😂

Hellowlol committed:
typing state.

Hellowlol committed:
Windows typing and doc strings

Hellowlol committed:
Add some docs to vehicle

Hellowlol committed:
add diff.

Hellowlol committed:
Add doc string to battery

all range is not formated with the unit the car uses.

Hellowlol committed:
charge docs and renaming some

propterties

Hellowlol committed:
Doc string and remove

and remove s for some properties as it not plural

Hellowlol committed:
Docstring sentry

Hellowlol committed:
Docstring software.py

Hellowlol committed:
kek

Hellowlol committed:
Update setup.py

Hellowlol committed:
fix session close.

Hellowlol commented:
Closing this as the progress is too slow. Ill maintain my own fork. Thanks anyway. 😃
